astropy/time/core.py

```diff
@@ -1592,7 +1592,23 @@ def __getattr__(self, attr):
                 cache[attr] = value
             return cache[attr]
 
-        elif attr in TIME_SCALES:  # allowed ones done above (self.SCALES)
+        elif '_' in attr:
```

Comment:
Maybe this should be a dunder so `mjd__float128` or whatever.

Reply:
It's meant to be user-visible, I think one underscore is plenty.

Reply:
I think the correct way to do that is to first check if the format as-is is known, then pull off the rightmost _thing and see if what's left is known, then see if the after-_ part is a known subformat. "is known" is just a "s in dict" check.

Reply:
The dunder conveys a (much-more) unambiguous separation between the attribute name and the subformat, both programmatically and to the human reader.  I think that `mjd_str` is a pretty good example of where this gets ambiguous on both counts since somebody might reasonably make their own format where MJD is represented as a string and call it `mjd_str`.  There is precedence, e.g. Django uses dunder in search queries to separate the name of the search field from the operation (https://docs.djangoproject.com/en/2.2/topics/db/search/).

Reply:
I followed @aarchiba's logic, first check whether it is known.

But I like @taldcroft's suggestion of a general `Time.to_value(fmt, subfmt)` even better - indeed if we default to `Quantity` for `TimeDelta`, as a side-effect that solves a long-standing annoyance of being able to do `.to(unit)` but not `to_value(unit)`.